Automattic/wp-calypso
# New Page Layout Picker: Convert new page layout modal to TypeScript

#### Changes proposed in this Pull Request

Adds types to the `@automattic/page-template-modal` package, which should make it easier to work on in the future.

The code is based off the redesign done in #49860 (instead of `trunk`) because it was much easier to type due to the fact that a lot of code got deleted in #49860.

Most of the changes are just changing type annotations, however there are a few small logic changes to deal with undefined/null values the compiler caught.

Note that it's still using the `transpile` command from `@automattic/calypso-build` to build the package. This does seem to work though, so perhaps babel has been configured to work with TypeScript. Ping ~sgomes~ @scinos to make sure this is the correct thing to do.

#### Testing instructions



* Build should succeed
* `yarn dev --sync` should still work, and the "new page layout modal" should still work correctly (sanity check, it shouldn't crash)
